app/models/mixins/purging_mixin.rb

```diff
@@ -43,6 +43,22 @@ module PurgingMixin
   extend ActiveSupport::Concern
 
   module ClassMethods
+    def purge_mode_and_value
+      [:date, purge_date]
+    end
+
+    def purge_timer
+      purge_queue(*purge_mode_and_value)
+    end
+
+    def purge_queue(mode, value)
+      MiqQueue.put(
```

Comment:
Purging is quick. We perform it as pure sql and do not download tons of records.

So the thought is there is no reason to limit which workers can purge.
But that is my though, wanted your feedback here.